### PR TITLE
Flesh out airline demo warehouse pipeline

### PR DESCRIPTION
**Pull request by mgasner**
screen shot 2018-12-06 at 12 45 54 pm

### CODE REVIEW

File: python_modules/airline-demo/airline_demo/pipelines.py [hunk 1]
```diff
@@ -66,21 +77,71 @@ def _db_load(data_frame, table_name, resources):
...
 
+
 def define_string_resource():
     return ResourceDefinition(
         resource_fn=lambda info: info.config,
         config_field=types.Field(types.String),
     )
 
+
+class TempfileManager(object):
```
Comment (schrockn) [inline python_modules/airline-demo/airline_demo/pipelines.py]: We're definitely going to want to add an option to not cleanup files for unittesting purposes. But we can add that when we get to it.

File: python_modules/airline-demo/airline_demo/solids.py [hunk 2]
```diff
@@ -40,25 +41,28 @@ def notebook_solid(name, inputs, outputs):
...
         table_name (str): THe name of the new table to create, if the materialization strategy
-            is 'table'.
+            is 'table'. Default: None.
+        inputs (list[InputDefinition]): Inputs, if any, for the new solid. Default: None.
 
     Returns:
         function:
             The new SQL solid.
     '''
+    if inputs is None:
```
Comment (schrockn) [inline python_modules/airline-demo/airline_demo/solids.py]: you can use:
```
      inputs = check.opt_list_param(inputs, 'inputs', InputDefinition)
```

File: python_modules/airline-demo/airline_demo/solids.py [hunk 3]
```diff
@@ -208,19 +215,22 @@ def download_from_s3(info):
...
             )
         )
         return target_path
 
-    if os.path.dirname(target_path):
+    if target_path and os.path.dirname(target_path):
         mkdir_p(os.path.dirname(target_path))
 
+    if target_path is None:
+        target_path = info.context.resources.tempfile.tempfile().name
```
Comment (schrockn) [inline python_modules/airline-demo/airline_demo/solids.py]: hmmm we might want to consider a more "functional" api here where the tempfile manager has a track less state

File: python_modules/airline-demo/airline_demo/solids.py [hunk 4]
```diff
@@ -612,11 +623,204 @@ def union_spark_data_frames(_info, left_data_frame, right_data_frame):
...
-    from q2_on_time_data
-    where origin='SFO'
+    from q2_sfo_outbound_flights
     ''',
     'table',
     table_name='average_sfo_outbound_avg_delays_by_destination',
+    inputs=[InputDefinition('q2_sfo_outbound_flights', dagster_type=SqlTableName)]
+)
+
+ticket_prices_with_average_delays = sql_solid(
```
Comment (schrockn) [inline python_modules/airline-demo/airline_demo/solids.py]: we should probably end up parameterizing this with a template

File: python_modules/airline-demo/airline_demo/pipelines.py (hunk 1 above)
Comment (mgasner) [reply python_modules/airline-demo/airline_demo/pipelines.py]: 👍

File: python_modules/airline-demo/airline_demo/solids.py (hunk 2 above)
Comment (mgasner) [reply python_modules/airline-demo/airline_demo/solids.py]: yep yep, i thought about this and my only question was whether we want to encourage users to use `check`, will make the change

File: python_modules/airline-demo/airline_demo/solids.py (hunk 3 above)
Comment (mgasner) [reply python_modules/airline-demo/airline_demo/solids.py]: yep, the thing i was trying to solve for here is the case where we want a named temporary file that persists across node execution boundaries. i'm very open to a more functional api that lets us keep the underlying NamedTemporaryFile around but i don't see the obvious change

File: python_modules/airline-demo/airline_demo/solids.py (hunk 4 above)
Comment (mgasner) [reply python_modules/airline-demo/airline_demo/solids.py]: totally, i think i want to reduce this machinery and the existing jinja-based stuff in the dagster_examples to a single reusable pattern

**Pull request merged**